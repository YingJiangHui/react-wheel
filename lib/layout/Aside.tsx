import React from 'react'
import {scopedClass} from '../helper/scopedClass';
const cn = scopedClass('makabaka-aside')
interface AsideProps {
 className?:string
}
 const Aside:React.FC<AsideProps> = ({children,className=''})=>{
  return (
    <div className={cn('',{extra:[className]})}>{children}</div>
  )
}
export default Aside