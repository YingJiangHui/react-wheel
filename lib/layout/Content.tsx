import React from 'react'
interface ContentProps {

}
const Content:React.FC<ContentProps> = ({children})=>{
  return (
    <div>{children}</div>
  )
}
export default Content