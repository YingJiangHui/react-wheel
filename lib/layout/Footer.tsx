import React from 'react'
interface FooterProps {

}
const Footer:React.FC<FooterProps> = ({children})=>{
  return (
    <div>{children}</div>
  )
}
export default Footer