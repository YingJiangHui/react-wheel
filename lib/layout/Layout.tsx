import React from 'react'
interface LayoutProps{

}
const Layout:React.FC<LayoutProps> = ({children})=>{
  return (
    <div>{children}</div>
  )
}
export default Layout