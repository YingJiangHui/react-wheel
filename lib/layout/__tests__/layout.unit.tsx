import renderer from "react-test-renderer";
import React from "react";
import Layout from "../Layout";
import Header from "../Header";
import Content from "../Content";
import Footer from "../Footer";
import Aside from "../Aside";

describe('Layout测试用例',()=>{
  it('上中下布局',()=>{
    const json = renderer.create(<Layout>
      <Header>Header</Header>
      <Content>Content</Content>
      <Footer>Footer</Footer>
    </Layout>).toJSON()
    expect(json).toMatchSnapshot()
  })
  
  it('上中下加上侧边栏',()=>{
    const json = renderer.create(<Layout>
      <Header>Header</Header>
      <Layout>
        <Aside>Aside</Aside>
        <Content>Content</Content>
      </Layout>
      <Footer>Footer</Footer>
    </Layout>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('上中下加上右边侧边栏',()=>{
    const json = renderer.create(<Layout>
      <Header>Header</Header>
      <Layout>
        <Content>Content</Content>
        <Aside>Aside</Aside>
      </Layout>
      <Footer>Footer</Footer>
    </Layout>).toJSON()
    expect(json).toMatchSnapshot()
  })
  
  it('上中下加侧边栏自适应高度',()=>{
    const json = renderer.create(<Layout>
      <Aside>Aside</Aside>
      <Layout>
        <Header>Header</Header>
        <Content>Content</Content>
        <Footer>Footer</Footer>
      </Layout>
    </Layout>).toJSON()
    expect(json).toMatchSnapshot()
  })
})