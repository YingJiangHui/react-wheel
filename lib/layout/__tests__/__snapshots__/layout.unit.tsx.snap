// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout测试用例 上中下加上侧边栏 1`] = `
<div
  className="makabaka-layout"
>
  <div
    className="makabaka-header"
  >
     
    Header
  </div>
  <div
    className="makabaka-layout-hasAside makabaka-layout"
  >
    <div
      className="makabaka-aside"
    >
      Aside
    </div>
    <div
      className="makabaka-content"
    >
      Content
    </div>
  </div>
  <div
    className="makabaka-footer"
  >
    Footer
  </div>
</div>
`;

exports[`Layout测试用例 上中下加上右边侧边栏 1`] = `
<div
  className="makabaka-layout"
>
  <div
    className="makabaka-header"
  >
     
    Header
  </div>
  <div
    className="makabaka-layout-hasAside makabaka-layout"
  >
    <div
      className="makabaka-content"
    >
      Content
    </div>
    <div
      className="makabaka-aside"
    >
      Aside
    </div>
  </div>
  <div
    className="makabaka-footer"
  >
    Footer
  </div>
</div>
`;

exports[`Layout测试用例 上中下加侧边栏自适应高度 1`] = `
<div
  className="makabaka-layout-hasAside makabaka-layout"
>
  <div
    className="makabaka-aside"
  >
    Aside
  </div>
  <div
    className="makabaka-layout"
  >
    <div
      className="makabaka-header"
    >
       
      Header
    </div>
    <div
      className="makabaka-content"
    >
      Content
    </div>
    <div
      className="makabaka-footer"
    >
      Footer
    </div>
  </div>
</div>
`;

exports[`Layout测试用例 上中下布局 1`] = `
<div
  className="makabaka-layout"
>
  <div
    className="makabaka-header"
  >
     
    Header
  </div>
  <div
    className="makabaka-content"
  >
    Content
  </div>
  <div
    className="makabaka-footer"
  >
    Footer
  </div>
</div>
`;
